sdk/program/src/stake/instruction.rs

```diff
@@ -261,6 +261,27 @@ pub enum StakeInstruction {
     ///   2. `[]` Reference vote account that has voted at least once in the last
     ///      `MINIMUM_DELINQUENT_EPOCHS_FOR_DEACTIVATION` epochs
     DeactivateDelinquent,
+
+    /// Redelegate activated stake to another vote account.
+    ///
+    /// Upon success:
+    ///   * the balance of the delegated stake account will be reduced to the rent exempt minimum,
+    ///     and scheduled for deactivation.
+    ///   * the provided uninitialized stake account will receive the original balance of the
+    ///     delegated stake account, minus the rent exempt minimum, and scheduled for activation to
+    ///     the provided vote account. Any existing lamports in the uninitialized stake account
+    ///     will also be included in the re-delegation.
+    ///
+    /// # Account references
+    ///   0. `[WRITE]` Delegated stake account to be redelegated. The account must be fully
+    ///       activated and carry a balance equal to twice the minimum delegation amount or greater
```

Comment:
The implementation does this differently, since it only leaves the rent-exempt reserve.  Ideally, only the rent-exempt reserve would stay in the source account, but that means we'd have an active stake account with less than the minimum delegation amount.  Maybe we force the minimum delegation + rent-exempt reserve to remain in the source account?

Reply:
Oops, stale doc. I did that originally. 

> Ideally, only the rent-exempt reserve would stay in the source account, but that means we'd have an active stake account with less than the minimum delegation amount. Maybe we force the minimum delegation + rent-exempt reserve to remain in the source account?

The source account is getting deactivated though, so I wanted to use the smallest amount of lamports possible. Ideally this would be zero, but not possible. It's ok to have an inactive stake account with less than the minimum delegation in it. Such an account can still be created by anybody at any time, and simply can't be delegated. I was envisioning that the client would immediately reclaim the lamports in the source account as soon as it deactivates

Reply:
Fair enough, WFM!